pkg/apis/kops/componentconfig.go

```diff
@@ -384,6 +384,21 @@ type KubeAPIServerConfig struct {
 	KubeletPreferredAddressTypes []string `json:"kubeletPreferredAddressTypes,omitempty" flag:"kubelet-preferred-address-types"`
 
 	StorageBackend *string `json:"storageBackend,omitempty" flag:"storage-backend"`
+
+	// The OpenID claim to use as the user name.
+	// Note that claims other than the default ('sub') is not guaranteed to be unique and immutable.
+	OIDCUsernameClaim *string `json:"oidcUsernameClaim,omitempty" flag:"oidc-username-claim"`
+	// If provided, the name of a custom OpenID Connect claim for specifying user groups.
```

Comment:
Mentioned to @justinsb on his pr, do we need testing?

Also any idea on documenting this stuff. 

Was not understand the data flow, and you are correct this will work ;)